File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i,e=0,len(s)-1
        strs=s.lower()
        def isalphanumeric(i):
            return (ord('a')<=ord(i)<=ord('z') or ord('0') <= ord(i) <= ord('9'))

        
        while(i<e):
            if (isalphanumeric(strs[i]))==False:
                i+=1
                continue
            if (isalphanumeric((strs[e])))==False:
                e-=1
                continue
            if strs[i]!=strs[e]:
                return False
            i+=1
            e-=1
        return True
        

File: 125-valid-palindrome/valid-palindrome.cs
public class Solution {
    public bool IsPalindrome(string s) {
        int start=0;
        int end=s.Length-1;

        s=s.ToLower();
        while(start<end){
            Console.WriteLine(s[start] + " " + s[end]);
            if(isAlphaNumeric(s[start])){
                if(isAlphaNumeric(s[end])){
                    if(s[start]==s[end]){
                        start++;
                        end--;
                    }else{
                        return false;
                    }
                }else{
                    end--;
                }
            }else{
                start++;
            }
        }
        return true;
    }

    public bool isAlphaNumeric(char c){
        //chained operation not allowed ('a'<=c<='z')
        // if(('a'<=c<='z') ||  ('A'<=c<='Z') || ('0'<=c<='9')){
        //     return true;
        // }

        //return char.IsLetterOrDigit(c)
        if((c>='a' && c<='z') || (c>='A' && c<='Z') || (c>='0' && c<='9')){
            return true;
        }
        return false;
    }
}